NRP-3606-Added endpoint for query wizard post

lgrzybowski opened the pull request:
I added post method for new query wizard.

lgrzybowski commented:
@jonathanchrisp @benSlaughter @leemcgeever 
I changed travis setting as new version of tins package (1.7.0) requires Ruby in version >= 2.0 
https://github.com/flori/tins

jonathanchrisp commented on lib/bwapi/client/brandwatch/usage_report.rb:
```diff
@@ -5,13 +5,13 @@ module Brandwatch
-        # @param clientId [Integer] Id of client
-        # @param startDate [DateTime] Report startDate
-        # @param endDate [DateTime] Report endDate
+        # @param client_id [Integer] Id of client
```
Good spot :+1:

jonathanchrisp commented on lib/bwapi/client/projects/queries.rb:
```diff
@@ -61,6 +61,32 @@ def create_query(project_id, opts = {})
+        # @option opts [Date] creationDate Date the query was created on
+        # @option opts [Date] lastModificationDate Modification date of the query
+        # @return [Hash] New query wizard
+        def create_query_wizard(project_id, opts = {})
```
could we rename this to `create_wizard_query`, is this going to replace the standard query?

lgrzybowski replied:
No we will have two entry point for building query, easy one and for more advanced users (current query builder).

jonathanchrisp replied:
Sounds good, yeah, I think `create_wizard_query` is better IMHO.

jonathanchrisp commented on lib/bwapi/client/projects/queries.rb:
```diff
@@ -61,6 +61,32 @@ def create_query(project_id, opts = {})
+        #
+        # @param project_id [Integer] Id of project
+        # @param opts [Hash] options hash of parameters
+        # @option opts [String] id Id of the query
```
We should rename `opts` to the appropriate parameter, please do this for the other instances below.

jonathanchrisp commented:
@lgrzybowski - out of interest what do we use the `tins` package for?

lgrzybowski commented:
@jonathanchrisp I changed the name and added two more endpoint as they are done now. I hope it can be merge as part of that PR. 
Regarding your question as I see tins is used by rubocop
```
rubocop (0.35.1)
      ....
      tins (<= 1.6.0)
```

leemcgeever commented on lib/bwapi/client/projects/queries.rb:
```diff
@@ -61,6 +70,32 @@ def create_query(project_id, opts = {})
+        #
+        # @param project_id [Integer] Id of project
+        # @param opts [Hash] options hash of parameters
+        # @option opts [Integer] id Id of the query
```
do we need `id` and `Id`?

leemcgeever commented on lib/bwapi/client/projects/queries.rb:
```diff
@@ -42,7 +51,7 @@ def get_query(project_id, query_id)
         # @param opts [Hash] options hash of parameters
-        # @option opts [String] id Id of the query
+        # @option opts [Integer] id Id of the query
         # @option opts [String] name Name of the project
```
do we need `id` and `Id`?

leemcgeever commented on lib/bwapi/client/projects/queries.rb:
```diff
@@ -85,6 +120,32 @@ def update_query(project_id, query_id, opts = {})
+        #
+        # @param project_id [Integer] Id of project
+        # @param opts [Hash] options hash of parameters
+        # @option opts [Integer] id Id of the query
```
do we need `id` and `Id`?

leemcgeever commented on lib/bwapi/client/projects/queries.rb:
```diff
@@ -85,6 +120,32 @@ def update_query(project_id, query_id, opts = {})
+        # @option opts [String] industry Industry of the query
+        # @option opts [String] usecase The Usecase the Query was created for
+        # @option opts [String] type Type of query
+        # @option opts [Object] queryWizardConfiguration JSON object containing configuration for Query Wizard
```
again  do we need type twice?

leemcgeever commented:
LGTM :+1:

jonathanchrisp commented:
LGTM - merging :+1:
